osu.Game/Screens/Play/Player.cs

```diff
@@ -640,57 +647,44 @@ public void Restart()
         /// <exception cref="InvalidOperationException">Thrown if this method is called more than once without changing state.</exception>
         private void updateCompletionState(bool skipStoryboardOutro = false)
         {
-            // screen may be in the exiting transition phase.
+            // If this player instance is already exiting upwards, don't attempt any kind of forward progress.
```

Comment:
The upwards/forward directions here are throwing me off some. Maybe this is better?

```suggestion
            // If this player instance is in the middle of an exit, don't attempt any kind of state update.
```